doc/sphinx/proofs/writing-proofs/proof-mode.rst

```diff
@@ -1,74 +1,176 @@
+definitions are shown in the form `n := 0 : nat`, for example, in which `nat` is the
+type of `0`.
 
-   Coq raises this error message when one attempts to use a proof editing command
-   out of the proof editing mode.
+The proof state when you begin proving a theorem shows
+the statement of the theorem below the line and no hypotheses above the
+line:
+
+.. coqtop:: none
+
+   Parameter P: nat -> Prop.
+
+.. coqtop:: out
+
+   Goal forall n m: nat, n > m -> P 1 /\ P 2.
+
+After applying the :tacn:`intros` :term:`tactic`, we see hypotheses above the line.
+The names of variables (`n` and `m`) and hypotheses (`H`) appear before a colon, followed by
+the type they represent.
+
+.. coqtop:: all
+
+   intros.
+
+Some tactics, such as :tacn:`split`, create new goals, which may
+be referred to as :gdef:`subgoals <subgoal>` in descriptions for clarity.
+Goals are numbered from 1 to N at each step of the proof to permit applying a
+tactic to specific goals.
+
+.. coqtop:: all
+
+   split.
+
+"Variables" may refer specifically to facts for which the type of their type
+is `Set` or `Type`, and "hypotheses" to :term:`propositions <proposition>`,
+which are facts for which the type of their type is `Prop` or `SProp`,
+but these terms are also used interchangeably.
+
+.. coqtop:: out
+
+   let t_n := type of n in idtac "type of n :" t_n;
+   let tt_n := type of t_n in idtac "type of" t_n ":" tt_n.
+   let t_H := type of H in idtac "type of H :" t_H;
+   let tt_H := type of t_H in idtac "type of" t_H ":" tt_H.
+
+A proof script, consisting of the tactics that are applied to prove a
+theorem, is often informally referred to as a "proof".
+The real proof, whether complete or incomplete, is a term, the :gdef:`proof term`,
+which users may occasionally want to examine.  (This is based on the
+*Curry-Howard isomorphism* :cite:`How80,Bar81,Gir89,H89`, which is
+a correspondence between between proofs and terms and between
+propositions and types of λ-calculus.  The isomorphism is also
+sometimes called the "propositions-as-types correspondence")
+
+The :cmd:`Show Proof` command displays the incomplete proof term
+before you've completed the proof.  For example:
+
+.. coqtop:: all
+
+   Show Proof.
+
+The incomplete parts, the goals, are represented by
+:term:`existential variables <existential variable>`
+with names that begin with `?Goal`.  The :cmd:`Show Existentials` command
+shows each existential with the hypotheses and conclusion for the associated goal.
+
+.. coqtop:: all
+
+   Show Existentials.
+
+Coq's kernel verifies the correctness of proof terms when it exits
+proof mode by checking that the theorem statement is the same as the
+type of the proof term.
+
+After a proof is completed, :cmd:`Print` `<theorem_name>`
+shows the proof term and its type.  The type appears after
+the colon (`forall ...`), as for this theorem from Coq's standard library:
+
+.. coqtop:: all
+
+   Print proj1.
```

Comment:
Did you notice this in the output?  I haven't had a disk drive for years.  Maybe worth rephrasing as "from the file system" or "from files" in a separate PR?  I don't quite see why this message is worth having, either, BTW.

![image](https://user-images.githubusercontent.com/1253341/103105540-54f27280-45e3-11eb-85dc-66c739538cfc.png)

Reply:
No, I had never noticed. Indeed, I agree that the message could be updated or even removed.